plugins/hydra_nevergrad_sweeper/hydra_plugins/hydra_nevergrad_sweeper/_impl.py

```diff
@@ -157,11 +159,32 @@ def sweep(self, arguments: List[str]) -> None:
             )
             self.validate_batch_is_legal(overrides)
             returns = self.launcher.launch(overrides, initial_job_idx=self.job_idx)
-            self.job_idx += len(returns)
             # would have been nice to avoid waiting for all jobs to finish
             # aka batch size Vs steady state (launching a new job whenever one is done)
+            self.job_idx += len(returns)
+            # check job status and prepare losses
+            failures = 0
+            to_tell: List[Tuple[ng.p.Parameter, float]] = []
             for cand, ret in zip(candidates, returns):
```

Comment:
What is the motivation for loop twice? (once to check the error and another time to tell the optimizer?)
can both be done in the same iteration?

Reply:
yes just split the logic but probably didnt help, so I updated it